src/software/networking/BUILD

```diff
@@ -13,10 +19,16 @@ cc_library(
 )
 
 cc_library(
-    name = "proto_multicast_sender",
-    hdrs = [
+    name = "threaded_proto_multicast_sender",
+    # We include the non-threaded class as srcs so the headers
+    # cannot accidentally be seen or included by other code
+    srcs = [
         "proto_multicast_sender.h",
-        "proto_multicast_sender.tpp",
+        "proto_multicast_sender.hpp",
```

Comment:
Why not just make proto_multicast_sender another target that private to this package?

Reply:
Because the headers would be transitively included in the main `threaded_...` target, so the `proto_multicast_sender.h` header would be visible to consumers and could be included. This is to protect against that.

Reply:
If that's overkill I can do `private` targets easily.

Reply:
I wouldn't mind changing tpp to hpp everywhere so it's all consistent

Reply:
@garethellis0 thoughts?

I think I'd prefer to revert to `.tpp` here and just make private targets because this is a pretty special case and I thought we had a decent reason for using `.tpp` over something else (mainly that it's super clear it's a special template implementation)

Reply:
I still perfer `.tpp` for the reasons that Mat mentioned here. I think just making it `private` is a fine solution here.

Reply:
Reverted to separate targets with `.tpp` files and made the non-threaded ones private.